packit_service/worker/checker/run_condition.py

```diff
@@ -0,0 +1,197 @@
+        self._actions_handler: Optional[ActionsHandler] = None
+
+    @property
+    def handler_kls(self):
+        if self._handler_kls is None:
+            logger.debug(f"Command handler: {self.service_config.command_handler}")
+            self._handler_kls = RUN_COMMAND_HANDLER_MAPPING[self.service_config.command_handler]
+        return self._handler_kls
+
+    @property
+    def working_dir(self) -> Optional[Path]:
+        if not self._working_dir:
+            if self.handler_kls == SandcastleCommandHandler:
+                path = (
+                    Path(self.service_config.command_handler_work_dir) / "run-condition-working-dir"
+                )
+                path.mkdir(parents=True, exist_ok=True)
+                self._working_dir = path
+            else:
+                self._working_dir = Path(tempfile.mkdtemp())
+            logger.info(
+                f"Created directory for the run-condition action: {self._working_dir}",
+            )
+        return self._working_dir
+
+    @property
+    def command_handler(self) -> CommandHandler:
+        if self._command_handler is None:
+            self._command_handler = self.handler_kls(
+                config=self.service_config,
+                working_dir=self.working_dir,
+            )
+        return self._command_handler
+
+    @property
+    def actions_handler(self) -> ActionsHandler:
+        if not self._actions_handler:
+            self._actions_handler = ActionsHandler(
+                self.job_config,
+                self.command_handler,
+            )
+        return self._actions_handler
+
+    def common_env(
+        self, version: Optional[str] = None, extra_env: Optional[dict[str, str]] = None
+    ) -> dict[str, str]:
+        env = self.job_config.get_package_names_as_env()
+        if version:
+            env["PACKIT_PROJECT_VERSION"] = version
+        if extra_env:
+            env.update(extra_env)
+        return env
+
+    def pre_check(self) -> bool:
+        project = self.project
+        git_ref = self.data.commit_sha
+        version = None
+
+        if self.data.event_type in (
+            github.pr.Action.event_type(),
+            pagure.pr.Action.event_type(),
+            gitlab.mr.Action.event_type(),
+            github.pr.Comment.event_type(),
+            pagure.pr.Comment.event_type(),
+            gitlab.mr.Comment.event_type(),
+            pagure.pr.Flag.event_type(),
+            github.check.PullRequest.event_type(),
+        ):
+            project = self.project.get_pr(int(self.data.pr_id)).source_project
+        elif self.data.event_type in (anitya.NewHotness.event_type(),):
+            project = self.service_config.get_project(
+                url=self.data.event_dict.get("distgit_project_url")
+            )
+            git_ref = self.data.tag_name
+            version = self.data.event_dict.get("version")
+        elif self.data.event_type in (
```

Comment:
this could be in theory also retrigger of `propose-downstream`, in which case the project should be probably the upstream one? I am not sure conditioning this only based on event type is the right way, but can't think of what would be better way for now (my first idea was based on the handler, but that would not be enough as well).

Reply:
> perhaps the event doesn't have `dist_git_project_url` set in `propose-downstream` case?

Thinking about it, that would make zero sense. So I suppose the only way is to parse the comment.

Reply:
> Thinking about it, that would make zero sense. So I suppose the only way is to parse the comment.

So could we use the information about handler so that we don't duplicate the parsing etc?

Reply:
In the end I decided to pass in task name as a string to avoid circular imports, WDYT?

Reply:
looks good 👍